CLAUDE.md

```diff
@@ -0,0 +1,321 @@
+
+### Core Architecture
+```
++-----------------+    +------------------+    +-----------------+
+|   Guest VM      |    |   Modelplex      |    |   Providers     |
+|                 |    |   (Host)         |    |                 |
+|  +-----------+  |    |  +-------------+ |    | +-------------+ |
+|  |    LLM    |  |    |  |   Model     | |    | |   OpenAI    | |
+|  |   Agent   |<-+----+->| Multiplexer |<+----+>|  Anthropic  | |
+|  +-----------+  |    |  +-------------+ |    | |   Ollama    | |
+|                 |    |  +-------------+ |    | +-------------+ |
+|                 |    |  |    MCP      | |    |                 |
+|                 |    |  | Integration |<+----+-> MCP Servers   |
+|                 |    |  +-------------+ |    |                 |
++-----------------+    +------------------+    +-----------------+
+        |                        |
+        +----modelplex.socket----+
+```
+
+## Technology Stack & Requirements
+
+### Core Technologies
+- **Language**: Go 1.24+ (currently 1.24.4 for security patches)
+- **Config**: TOML (not YAML) with environment variable substitution
+- **CLI**: jessevdk/go-flags for professional CLI interface
+- **HTTP Router**: gorilla/mux for OpenAI-compatible API
+- **Testing**: testify framework with comprehensive test coverage
+- **Logging**: structured logging with slog (not standard log package)
+
+### Key Dependencies
+```go
+require (
+    github.com/gorilla/mux v1.8.1
+    github.com/jessevdk/go-flags v1.5.0
+    github.com/pelletier/go-toml/v2 v2.1.1
+    github.com/stretchr/testify v1.8.4
+)
+```
+
+## Code Architecture & Components
+
+### Directory Structure
+```
+cmd/modelplex/           # CLI application entry point
+internal/
+├── config/              # TOML configuration management
+├── multiplexer/         # Model routing and provider selection
+├── providers/           # AI provider implementations (OpenAI, Anthropic, Ollama)
+├── proxy/               # OpenAI-compatible API proxy
+├── server/              # Unix socket HTTP server
+├── mcp/                 # Model Context Protocol integration
+└── monitoring/          # Structured logging utilities
+test/
+├── integration/         # Full system integration tests
+└── testutil/           # Test helper utilities
+```
+
+### Provider Implementations
+
+#### OpenAI Provider
+- Full OpenAI API compatibility
+- Bearer token authentication
+- Direct passthrough for requests/responses
+- Supports chat completions and text completions
+
+#### Anthropic Provider
+- x-api-key header authentication
+- anthropic-version: "2023-06-01" header required
+- System message transformation (moved to separate field)
+- Response format normalization to OpenAI-compatible
+
+#### Ollama Provider
+- No authentication required
+- Local inference endpoints: /api/chat, /api/generate
```

Comment:
I think I changed this to `/model/v1/...`, do we want to change it back? I'm thinking `/model/*` `/mcp/*` `/_host/*` endpoints